app/services/api/teachers/action.rb

```diff
@@ -0,0 +1,63 @@
+module API::Teachers
+  class Action
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    attribute :author
+    attribute :lead_provider_id
+    attribute :participant_id
+    attribute :course_identifier
+
+    validates :lead_provider_id, presence: { message: "Your update cannot be made as the '#/lead_provider' is not recognised. Check lead provider details and try again." }
+    validates :participant_id, presence: { message: "The property '#/participant_id' must be present" }
+    validates :course_identifier,
+              inclusion: { in: %w[ecf-mentor ecf-induction],
+                           message: "The entered '#/course_identifier' is not recognised for the given participant. Check details and try again." },
+              presence: { message: "Enter a '#/course_identifier' value for this participant." },
+              allow_blank: false
+    validate :teacher_exists
+    validate :training_period_exists
+
+    def teacher
+      @teacher ||= Query.new(lead_provider_id:).teacher_by_api_id(participant_id)
```

Comment:
The controller will be doing this lookup/safety check anyway so I think we can keep it to a simple `Teacher.find_by` as in the `Withdraw` service https://github.com/DFE-Digital/register-early-career-teachers-public/pull/1569/files#diff-3066a5a6fe15f8cfbb6c56873b9f49a1d01e990a38564cde7f4e14dc7576fb93R51